packages/contracts/contracts/libs/v0.8.x/minter-libs/DALib.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title Art Blocks Merkle Library
+ * @notice This library is designed to manage and verify merkle based gating for Art Blocks projects.
+ * It provides functionalities such as updating the merkle root of project, verifying an address against a proof,
+ * and setting the maximum number of invocations per address for a project.
+ * @author Art Blocks Inc.
+ */
+
+library DALib {
+    using SafeCast for uint256;
+
+    struct DAProjectConfig {
```

Comment:
WDYT about just splitting this library up into a liner and exponential version? I find it a little odd to have both `timestampEnd` and `priceDecayHalfLifeSeconds` defined in the same struct, and every function in this library is either only used by Exp or Lin. Might be safer/cleaner to just split the library?

Reply:
agreed. initially was holding out some hope for more shared functionality across the actual functions, but now that the minters are finalizing I think its safe/cleaner to split them